sstables/kl/reader_impl.hh

```diff
@@ -334,32 +335,36 @@ private:
                 break;
             }
         case state::CELL_VALUE_BYTES:
-            if (read_bytes_contiguous(data, _u32, _val) != read_status::ready) {
+            if (read_bytes(data, _u32, _val_fragmented) != read_status::ready) {
                 _state = state::CELL_VALUE_BYTES_2;
                 break;
             }
         case state::CELL_VALUE_BYTES_2:
         {
             row_consumer::proceed ret;
             if (_deleted) {
+                _val = temporary_buffer<char>(4);
+                auto v = fragmented_temporary_buffer::view(_val_fragmented);
+                read_fragmented(v, 4, reinterpret_cast<bytes::value_type*>(_val.get_write()));
```

Comment:
I think you need to move this below the size check.

Reply:
(and change the size check to reference _val_fragmented, right now it can be optimized out since _val is created with a size of 4).

Reply:
Ok, fixed in the update